atc/worker/worker.go

```diff
@@ -463,6 +470,92 @@ func (worker *gardenWorker) createVolumes(logger lager.Logger, isPrivileged bool
 	return volumeMounts, nil
 }
 
+func (worker *gardenWorker) cloneLocalVolumes(
+	logger lager.Logger,
+	teamID int,
+	privileged bool,
+	container db.CreatingContainer,
+	locals []mountableLocalInput,
+) ([]VolumeMount, error) {
+	mounts := make([]VolumeMount, len(locals))
+
+	for i, localInput := range locals {
+		inputVolume, err := worker.volumeClient.FindOrCreateCOWVolumeForContainer(
+			logger,
+			VolumeSpec{
+				Strategy:   localInput.desiredCOWParent.COWStrategy(),
+				Privileged: privileged,
+			},
+			container,
+			localInput.desiredCOWParent,
+			teamID,
+			localInput.desiredMountPath,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		mounts[i] = VolumeMount{
+			Volume: inputVolume,
+			MountPath: localInput.desiredMountPath,
+		}
+	}
+
+	return mounts, nil
+}
+
+func (worker *gardenWorker) cloneRemoteVolumes(
+	logger lager.Logger,
+	teamID int,
+	privileged bool,
+	container db.CreatingContainer,
+	nonLocals []mountableRemoteInput,
+) ([]VolumeMount, error) {
+	mounts := make([]VolumeMount, len(nonLocals))
+
+	var g errgroup.Group
```

Comment:
Can we pass the context from `FindOrCreateContainer` down to this level so that we can create a group with context? E.g. ` g, ctx := errgroup.WithContext(ctx)`. This would allow us to cancel streaming if `FindOrCreateContainer` is cancelled.

Reply:
it looks like we don't pass contexts to FindOrCreateVolumes, and the context that errgroup.WithContext returns should be used by funcs that we want to cancel anytime the parent context cancels. Even StreamTo doesn't take in a context. I might be misunderstanding how this works but I don't think anything will be canceled unless we use the returned ctx.

Reply:
I added a comment about using a context to #3928 since we should probably do that for both container and volume creation. Let's not do it for this issue.